### PR TITLE
[.Net] fix #2859

### LINKED ISSUE #2859: [.Net][Feature Request]: Sample update - Add getting-start samples for BasicSample project
### Is your feature request related to a problem? Please describe.

`AutoGen.BasicSample` project used to contain getting-start examples for AutoGen.Net. Currently it has the following misalignments which makes it harder to follow for starters.
- some examples not "basic sample". For example, Example 4 is hard to follow
- some examples are placed in the wrong order. For example `Example 5` should go beyond `Example 4` for a better learning curve

~In the next milestone (0.0.15), we should improve `AutoGen.BasicSample` from the following three aspects~
~- keep the essential examples and remove the rest: only keep the examples that are necessary, removing model-specific examples into their own example projects.~
~- easy-to-follow learning path: re-organize the order of samples to make it more friendly to AutoGen beginner~
~- easy-to-follow example: replace complicated example to simple one~

Therefore, we should improve the `AutoGen.BasicSample` by only keeping the actual `getting-start` samples. For the rest of samples, either delete them or move to specific agent sample project (for example, `Example08_LMStudio should be moved to `AutoGen.LMStudio.Sample`)

Here is the sample with a more intuitive order to be kept in basic samples.
- [x] Chat with one agent (text message + multi-modal message)
- [x] Use tool call in agent (self-invoke + invoke by others)
- [x] Use middleware to extend agent behavior
  - [x] token count
  - [x] rag sample (what's date today)
- [x] GroupChat
  - [x] coder and commenter 
- [x] Getting more control in group chat (FSM)
  - [x] fill a form 

### Describe the solution you'd like

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by LittleLittleCloud**
## Related issue number

fix #2859

The update on website will come later

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by LittleLittleCloud** [comment]
@teeshajain17

**Review by DavidLuong98 — APPROVED** [review]
